### PR TITLE
update: added check to handle error

### LINKED ISSUE #559: Connecting with Argent web wallet on mainnet gives an application side error
Capture d’écran 2024-03-21 à 15 01 41
Capture d’écran 2024-03-21 à 15 02 02

Tried to connect on https://starknet.quest/quest/107 so maybe a `.?` missing on the quest page, it should take 10-20 minutes to find.

### PR DESCRIPTION
**Pull request by raizo07**
# Pull Request type

Added a check to avoid the errors related to trying to access properties of an undefined or null `data` object.

Also added a check for `data` before attempting to access the 'error' property to prevent the TypeError.

closes https://github.com/starknet-id/starknet.quest/issues/559

### CODE REVIEW

**Comment by ayushtom** [comment]
closes #559

File: context/QuestsProvider.tsx [hunk 1]
```diff
@@ -58,7 +58,11 @@ export const QuestsContextProvider = ({
   useMemo(() => {
     (async () => {
       const data: GetQuestsRes = await getQuests();
-
+      
+      if (!data) {
+        // Handle the case when data is null or undefined, maybe by setting a default value or returning early
+        return;
+    }
```
Comment (fricoben) [inline context/QuestsProvider.tsx]: The error is not here but in the `[questPage]` part

To delete

File: context/QuestsProvider.tsx [hunk 2]
```diff
@@ -119,7 +123,8 @@ export const QuestsContextProvider = ({
     getCompletedBoosts(hexToDecimal(address)).then(
       (data: number[] | QueryError) => {
         if ((data as QueryError).error) return;
-        setCompletedBoostIds(data as number[]);
+        if (data && (data as QueryError).error) return; // Check if 'data' exists before accessing 'error'
```
Comment (fricoben) [inline context/QuestsProvider.tsx]: The error is not here but in the `[questPage]` part

To delete

**Review by fricoben — CHANGES_REQUESTED** [review]
The bug is not fixed. 

To reproduce the bug you simply have to:
- [ ] Visit https://starknet.quest/quest/107
Capture d’écran 2024-04-03 à 13 56 41

- [ ] Connect to *Argent Web Wallet* (by Email)

**Comment by ayushtom** [comment]
@raizo07 could you fix this? i think we will have to reassign if this soon if you haven't been able to fix this yet

**Comment by raizo07** [comment]
I just got some pointers from @Fricoben and I'm working on it already.

**Comment by ayushtom** [comment]
@raizo07 i am unassigning this issue from you. This has taken a long time and i dont see much progress. You can continue to work and fix this but this won't be considered for your OD rewards.

**Comment by ayushtom** [comment]
this pr is not fixing the issue as of now so closing

**Pull request closed without merging**